lib/iris/_merge.py

```diff
@@ -1231,14 +1231,18 @@ def merge(self, unique=True):
                 if is_lazy_data(data):
                     all_have_data = False
                 else:
-                    data = as_lazy_data(data)
+                    if isinstance(data, np.ma.MaskedArray):
```

Comment:
@marqh This should be `ma.MaskedArray` not `np.ma.MaskedArray` to adhere to the previous namespace usgage